lib/explorer/data_frame.ex

```diff
@@ -129,17 +129,65 @@ defmodule Explorer.DataFrame do
 
   @doc """
   Reads a IPC file into a dataframe.
+
+    ## Options
+
+      * `columns` - List with name of columns to be selected. Defaults to all columns.
+      * `projection` - List with the index of columns to be selected.
   """
   @spec read_ipc(filename :: String.t()) :: {:ok, DataFrame.t()} | {:error, term()}
-  def read_ipc(filename), do: Explorer.PolarsBackend.DataFrame.read_ipc(filename)
+  def read_ipc(filename, opts \\ []) do
+    opts =
+      keyword!(opts,
+        columns: nil,
+        projection: nil
+      )
+
+    backend = backend_from_options!(opts)
+
+    backend.read_ipc(
+      filename,
+      opts[:columns],
+      opts[:projection]
+    )
+  end
+
+  @doc """
+  Similar to `read_ipc/2` but raises if there is a problem reading the IPC file.
+  """
+  @spec read_ipc!(filename :: String.t(), opts :: Keyword.t()) :: DataFrame.t()
+  def read_ipc!(filename, opts \\ []) do
+    case read_ipc(filename, opts) do
+      {:ok, df} -> df
+      {:error, error} -> raise "#{error}"
+    end
+  end
 
   @doc """
   Writes a dataframe to a IPC file.
+  Apache IPC is a language-agnostic columnar data structure that can be used to store data frames.
```

Comment:
I would only add a line here. It's common to separate the ["title"](https://hexdocs.pm/elixir/writing-documentation.html#recommendations) from the body of the doc.

```suggestion

  Apache IPC is a language-agnostic columnar data structure that can be used to store data frames.
```